apps/vaporgui/NavigationUtils.cpp

```diff
@@ -0,0 +1,403 @@
+    for (int i = 0; i < winNames.size(); i++) {
+        vector<string> dataSetNames = dataStatus->GetDataMgrNames();
+        for (int j = 0; j < dataSetNames.size(); j++) {
+            vector<RenderParams *> rParams;
+            paramsMgr->GetRenderParams(winNames[i], dataSetNames[j], rParams);
+
+            if (!rParams.size()) continue;
+
+            size_t local_ts = dataStatus->MapGlobalToLocalTimeStep(dataSetNames[j], ts);
+
+            for (int k = 0; k < rParams.size(); k++) {
+                RenderParams *rp = rParams[k];
+                rp->SetCurrentTimestep(local_ts);
+
+                // If variable was not initialzed but now is, reset the TF mapping range
+                std::string varNames[2] = {rp->GetVariableName(), rp->GetColorMapVariableName()};
+
+                for (int l = 0; l < 2; l++) {
+                    if (varNames[i].empty()) continue;
+
+                    MapperFunction *tf = rp->GetMapperFunc(varNames[i]);
+                    vector<double>  range = tf->getMinMaxMapValue();
+                    if (abs(range[1] - range[0]) > FLT_EPSILON) continue;
+
+                    DataMgr *dm = dataStatus->GetDataMgr(dataSetNames[j]);
+                    if (!dm) continue;
+
+                    if (!dm->VariableExists(local_ts, varNames[i], 0, 0)) continue;
+
+                    if (dm->GetDataRange(local_ts, varNames[i], 0, 0, range) < 0) continue;
+
+                    bool SSE = paramsMgr->GetSaveStateEnabled();
+                    paramsMgr->SetSaveStateEnabled(false);
+                    tf->setMinMaxMapValue(range[0], range[1]);
+                    paramsMgr->SetSaveStateEnabled(SSE);
+                }
+            }
+        }
+    }
+
+    paramsMgr->EndSaveStateGroup();
+}
+
+
+long NavigationUtils::GetCurrentTimeStep(VAPoR::ControlExec *ce)
+{
+    AnimationParams *aParams = (AnimationParams *)ce->GetParamsMgr()->GetParams(AnimationParams::GetClassType());
+    VAssert(aParams);
+
+    return aParams->GetCurrentTimestep();
+}
+
+
+VAPoR::ViewpointParams *NavigationUtils::GetActiveViewpointParams(ControlExec *ce)
+{
+    ParamsMgr *      pm = ce->GetParamsMgr();
+    auto             activeViz = GetGUIStateParams(ce)->GetActiveVizName();
+    ViewpointParams *vp = pm->GetViewpointParams(activeViz);
+    return vp;
+}
+
+
+GUIStateParams *NavigationUtils::GetGUIStateParams(ControlExec *ce)
+{
+    ParamsMgr *pm = ce->GetParamsMgr();
+    return ((GUIStateParams *)pm->GetParams(GUIStateParams::GetClassType()));
+}
+
+
+AnimationParams *NavigationUtils::GetAnimationParams(ControlExec *ce) { return ((AnimationParams *)ce->GetParamsMgr()->GetParams(AnimationParams::GetClassType())); }
+
+
+
+// This is some code that someone decided is worth keeping for future reference.
+// It allegedly worked at some point.
+// I don't want to have to write any camera code without refactoring the entire
+// system so I am leaving it here in case the functionality is needed again.
```

Comment:
@clyne is it ok for us to remove VAPOR3_0_0_ALPHA code at this point?

Reply:
I'd say so. We still have lot of detritus like this throughout the code base that needs to be clean up. But it seems reasonable to remove this now.

Reply:
I would like to leave this because, as the comment says, the camera code is quite a mess and I think we should keep it around until the camera backend as a whole is refactored. I can change the `ifdef` title if that makes its purpose more clear.

Reply:
Ok, let's not worry about it now then.